HipercowApi/Tools/ClusterLoadQuery.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Imperial College London. All rights reserved.
+
+namespace HipercowApi.Tools
+{
+    using HipercowApi.Models;
+    using Microsoft.AspNetCore.SignalR;
+    using Microsoft.Hpc.Scheduler;
+    using Microsoft.Hpc.Scheduler.Properties;
+
+    /// <summary>
+    /// The implementation of IClusterLoadQuery, used to
+    /// query a cluster headnode for its current load.
+    /// </summary>
+    public class ClusterLoadQuery : IClusterLoadQuery
+    {
+        /// <inheritdoc/>
+        public ClusterLoad GetClusterLoad(string cluster, IScheduler scheduler)
+        {
+            var nodeLoads = new List<NodeLoad>();
+            var filter = Utils.GetFilterNonComputeNodes(cluster);
+            var nodeList = scheduler.GetNodeList(filter, null);
+
+            foreach (ISchedulerNode node in nodeList)
+            {
+                var cores = node.GetCores();
+                int busy = 0;
+
+                foreach (ISchedulerCore core in cores)
+                {
+                    busy += (core.State == SchedulerCoreState.Busy) ? 1 : 0;
+                }
```

Comment:
I think you mentioned this could probably be made terser. Could you do something like  
```suggestion
                int busy = cores.Where(c => c.State == SchedulerCoreState.Busy).Count;
```

Reply:
Alas, cores doesn't have a method "Where" - I've wrestled with this one a bit as it looks like it should be simplifiable, but nothing has worked yet. cores is an `ISchedulerCollection` in HPC, which doesn't seem to support any useful enumeration methods.

Reply:
that's a shame!